### PR TITLE
[BEAM-6489] Remove usages of a deprecated flag and simplify SDK dependencies' version spec.

### PR DESCRIPTION
**Pull request by tvalentyn**
This change:
- Adds the dependency to newly released dill==0.2.9, which removes the necessity to pass --process-dependency-links to pip, since we no longer need to pick up an unreleased change.
- Cleans up tox.ini to use platform-specific dependecy declaration https://setuptools.readthedocs.io/en/latest/setuptools.html#declaring-platform-specific-dependencies 
- Sorts the SDK dependencies alphabetically.

Post-Commit Tests Status (on master branch)
------------------------------------------------------------------------------------------------

Lang | SDK | Apex | Dataflow | Flink | Gearpump | Samza | Spark
--- | --- | --- | --- | --- | --- | --- | ---
Go | [Build Status](https://builds.apache.org/job/beam_PostCommit_Go/lastCompletedBuild/) | --- | --- | --- | --- | --- | ---
Java | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Apex/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Dataflow/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Flink/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Java_PVR_Flink_Batch/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Java_PVR_Flink_Streaming/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Gearpump/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Samza/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Spark/lastCompletedBuild/)
Python | [Build Status](https://builds.apache.org/job/beam_PostCommit_Python_Verify/lastCompletedBuild/) | --- | [Build Status](https://builds.apache.org/job/beam_PostCommit_Py_VR_Dataflow/lastCompletedBuild/) </br> [Build Status](https://builds.apache.org/job/beam_PostCommit_Py_ValCont/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Python_VR_Flink/lastCompletedBuild/) | --- | --- | ---

### CODE REVIEW

**Comment by tvalentyn** [comment]
Run Portable_Python PreCommit

**Comment by tvalentyn** [comment]
Run Portable_Python PreCommit

**Comment by tvalentyn** [comment]
Run Portable_Python PreCommit

**Comment by tvalentyn** [comment]
CC'ing a few committers - feel free to merge this to unblock the Python Precommit suite. One review is enough. Thank you. 
@mxm @robertwb @aaltay @charlesccychen

**Comment by charlesccychen** [comment]
Thank you Valentyn!  This is much cleaner and unbreaks precommits.

**Comment by mxm** [comment]
Thank you for taking care of this!

File: sdks/python/setup.py [hunk 1]
```diff
@@ -101,34 +100,29 @@ def get_version():
...
+    'futures>=3.1.1,<4.0.0',
     'grpcio>=1.8,<2',
     'hdfs>=2.1.0,<3.0.0',
     'httplib2>=0.8,<=0.11.3',
     'mock>=1.0.1,<3.0.0',
     'oauth2client>=2.0.1,<4',
     # grpcio 1.8.1 and above requires protobuf 3.5.0.post1.
     'protobuf>=3.5.0.post1,<4',
+    # pyarrow is not supported on Windows for Python 2 [BEAM-6287]
+    ('pyarrow>=0.11.1,<0.12.0; python_version >= "3.0" or '
```
Comment (udim) [inline sdks/python/setup.py]: Thank you for this cleanup!

**Comment by tvalentyn** [comment]
cc: @RobbeSneyders

**Pull request merged**